resources/models/resource.py

```diff
@@ -132,6 +132,16 @@ class ResourceQuerySet(models.QuerySet):
     def visible_for(self, user):
         if is_general_admin(user):
             return self
+        elif is_unit_admin(user):
+            administered_unit_ids = Unit.objects.administered_by(user).values_list('id', flat=True)
+            administered_resources_q = Q(unit__in=administered_unit_ids)
+            public_resources_q = Q(public=True)
+            return self.filter(administered_resources_q | public_resources_q)
+        elif is_unit_manager(user):
+            managed_unit_ids = Unit.objects.managed_by(user).values_list('id', flat=True)
+            managed_resources_q = Q(unit__in=managed_unit_ids)
+            public_resources_q = Q(public=True)
+            return self.filter(managed_resources_q | public_resources_q)
```

Comment:
Shouldn't these both cases be combined, i.e. if the user is admin for some units and manager for some others?

I don't think it's useful to do the `is_unit_admin` or `is_unit_manager` calls, since they will only cause unneeded SQL queries.

Also, it's not good to pull the IDs to Python, but rather do the query with SQL as a whole.

I would do it like this (untested though):
```python
def visible_for(self, user):
    if is_general_admin(user):
        return self
    in_admined_units = Q(unit__in=Unit.objects.administered_by(user))
    in_managed_units = Q(unit__in=Unit.objects.managed_by(user))
    is_public = Q(public=True)
    return self.filter(in_admined_units | in_managed_units | is_public)
```

Reply:
hmm.. but now that I look again at the code of `UnitQuerySet.managed_by`, I see that (general or unit) administrators are already considered there, so it should be enough to do it like this (without even adding the `administered_by` method to the UnitQuerySet):
```python
def visible_for(self, user):
    if is_general_admin(user):
        return self
    in_managed_units = Q(unit__in=Unit.objects.managed_by(user))
    is_public = Q(public=True)
    return self.filter(in_managed_units | is_public)
```